src/Factory/KeyPairFactory.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/*
+ * This file is part of the ACME PHP library.
+ *
+ * (c) Titouan Galopin <galopintitouan@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace AcmePhp\Ssl\Factory;
+
+use AcmePhp\Ssl\KeyPair;
+use AcmePhp\Ssl\PrivateKey;
+use AcmePhp\Ssl\PublicKey;
+
+/**
+ * Crate KeyPair entities from PEMs.
+ *
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ */
+class KeyPairFactory
```

Comment:
Why don't you like new? Generally speaking, the simpler is usually the better, creating a factory like this one seems overkill IMO.